### DISCUSSION TITLE
Reloading a scene as a trial in Unity

### DISCUSSION THREAD

**Discussion by BearsBeetsBattlestarGalactica** [immersivecognition/unity-experiment-framework#156] [Q&A] [answered]
Hello together!

Thank you for making something like the UXF possible!

Right now I am working on an experiment where the subject has to navigate through a scene and move objects. After a trial I want to reset the whole scene (all objects to their starting positions) and the next trial starts.
Theoretically there is a function that does that. The problem is that the starting UXF interface (with age etc.) is also being reset by using the function. 

Do you have an Idea on how to reset the scene without saving every objects position and rotation?

Kind regards
Dominic

**Comment by jackbrookes** [comment, marked as answer]
Hey!

Probably the easiest way for you is to reload the scene. When you reload the scene and you want to keep the UXF Session, you can enable "Don't destroy on load new scene" and "Set as main instance"

https://github.com/immersivecognition/unity-experiment-framework/wiki/Multi-scene-experiments

If you have things like trackers, you will need to make those persist the reload too DontDestroyOnLoad 

https://docs.unity3d.com/ScriptReference/Object.DontDestroyOnLoad.html

**Reply by BearsBeetsBattlestarGalactica** [reply]
Hello Jack,

This framework saves lifes (and countless hours in the office )!:D
Thank you for your answer. It worked like a charm. I also watched your youtube tutorial and it was super helpful. I know that there is a written documentation but video tutorials on coding are unbeatable.

Are there any other video tutorials from you about the UXF framework?

Kind regards 
Dominic
